tests/console/validate_selected_patterns.pm

```diff
@@ -0,0 +1,51 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validation module to check patterns installed.
+# Scenarios covered:
+# - zypper search for named patterns(base,enhanced_base;
+# - zypper search for removed pattern apparmor
+#
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils;
+
+sub run {
+
+    select_console('root-console');
+
+    my @installed_patterns = split /,/, get_var('PATTERNS');
+    my %patterns           = (
+        base     => 'base',
+        minimal  => 'enhanced_base',
+        apparmor => 'apparmor',
+        x        => 'x11',
+        yast2    => 'yast2_basis'
+        );
+
+    foreach my $pattern (@installed_patterns) {
+        if !exists($patterns{$pattern}) {
```

Comment:
This will trigger die for each pattern to deselect (e.g. `-base`).

Reply:
@jknphy had a very nice suggestion that i pull.